src/tool_operate.c

```diff
@@ -1940,12 +1940,8 @@ static CURLcode single_transfer(struct GlobalConfig *global,
 
         /* open file for reading: */
         FILE *file = fopen(config->etag_compare_file, FOPEN_READTEXT);
-        if(!file && !config->etag_save_file) {
-          errorf(global,
-                 "Failed to open %s", config->etag_compare_file);
-          result = CURLE_READ_ERROR;
-          break;
-        }
+        if(!file)
+          warnf(global, "Failed to open %s", config->etag_compare_file);
```

Comment:
consider emitting the error code; this can mask permission errors, malformed paths, etc.

Reply:
You mean including some explanation in the warning message? Getting that human readable and understandable is not easy.

Reply:
At the very least `strerror(errno)` is helpful.

Reply:
will add!